### ISSUE TITLE
Credo Appeasement: Refactoring

### ISSUE THREAD

**Issue by crertel** [boydm/scenic#67]
This is about clearing up all the refactoring opportunities that Credo complains about.

**Comment by crertel** [comment]
So, working through this so far, a huuuuuge chunk were complaints about pipelines not starting with raw values. I disabled that, as that's one of the more esoteric and annoying credo complaints. Consider an example case like:

```
vector_add(a, b)
|> vector_cross(c)
|> vector_print()
```

Credo by default would prefer:


```
a
|> vector_add(b)
|> vector_cross(c)
|> vector_print()
```

So, that's off now.

There are a number of nesting complaints that were disabled in-situ, mostly because the refactor I think would've been harder (due to missing docs) and more error-prone (due to missing tests). We probably want to revisit those later.

The thing I'm grinding through now is a bunch of [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) warnings. Some of these I think I can fix by breaking things out into their own functions, but I might have to just tell Credo to ignore them in places where that's not easy to do.

Overall, I think we'll end up with some pragma'd complaints disabled for credo but we can at least stop new problems from coming in--and then we can fix those other issues as we have time.

**Comment by crertel** [comment]
Woohoo, fix open. See #73

**Comment by crertel** [comment]
Resolved by #73, closing.
